bots/goodbot.py

```diff
@@ -278,17 +279,69 @@ def process(self, msg):
 					"topic": topic,
 					"content": f"{response} Need some help! :point_up:"
 				})
-			elif "goodbot" in content and content[0] != "!help":
+
+			def config():
+				response = self.replies["confighelp"]
+				if sender_email not in self.config["botadmins"]:
+					response = "You are not authorized to make this action."
+				else:
+					try:
+						if content[1].lower() == "view":
+							response = f"```json\n{json.dumps(self.config, indent=2)}\n```"
+						elif content[1].lower() == "update":
+							key, value = content[2].split(":", maxsplit=1)
+							if key in self.config:
+								try:
+									copy = self.config.copy()
+									copy[key] = json.loads(value)
+									json.loads(json.dumps(copy))
+									self.config[key] = json.loads(value)
+									response = "Configuration updated successfully."
+								except Exception as e:
+									response = f"Input is not valid JSON. {e}"
+							else:
+								response = "Key does not exist in configuration."
+						elif content[1].lower() == "commit":
+							cmds = [
+								"git add config/config.json",
+								f"git commit -m {content[2]} --author={sender_email}",
+								"git push origin --dry-run"
+							]
+							with open(Path(__file__).parents[1].joinpath("config", "config.json"), "w") as file:
+								json.dump(self.config, file, indent="\t")
+							for cmd in cmds:
+								subprocess.run(shlex.split(cmd))
+							response = "Committed to repository."
+						elif content[1].lower() == "reset":
+							with open(Path(__file__).parents[1].joinpath("config", "config.json")) as file:
+								self.config = json.load(file)
+							response = "Configuration reset successfully."
+					except Exception:
+						pass
+				self.client.send_message({
+					"type": message_type,
+					"to": destination,
+					"topic": topic,
+					"content": f"{response}"
+				})
+
+			if "goodbot" in content and content[0].lower() != "!help":
 				self.client.send_message({
 					"type": message_type,
 					"to": destination,
 					"topic": topic,
 					"content": f"{greeting} :blush: What can I do for you today?"
 				})
-			else:
-				return
-		except Exception as e:
-			print(e)
+
+			keywords = {
```

Comment:
Nice!

Reply:
To make this more readable and easy to add if things become plenty, put each key word and value pair on it's own line and use the almighty sorting to keep it clean 👍

Reply:
Fixed in 3a900185233bed1bd2210328232ef1481be60557